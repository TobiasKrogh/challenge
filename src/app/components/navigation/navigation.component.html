<nav>
  <a
    *ngFor="let target of navigationTargets"
    class="navigation-link"
    [routerLink]="target.link"
    routerLinkActive="{{ target.ref }}--active"
  >
    <img
      src="/assets/images/navigation/{{ target.ref }}.v1.svg"
      alt="{{ target.ref }}"
    />
  </a>
</nav>
