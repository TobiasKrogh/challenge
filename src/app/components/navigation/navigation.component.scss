:host {
  display: block;
  width: var(--navigation-width);
}

nav {
  align-items: center;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  margin-top: 2.5rem;
}

.navigation-link {
  display: block;
  height: 1.5rem;
  width: 1.5rem;
}

.calculations {
  background-image: url(/assets/images/navigation/calculations.v1.svg);
}

.apps {
  background-image: url(/assets/images/navigation/apps.v1.svg);
}

.external {
  background-image: url(/assets/images/navigation/external.v1.svg);
}

.reports {
  background-image: url(/assets/images/navigation/reports.v1.svg);
}

.logout {
  background-image: url(/assets/images/navigation/logout.v1.svg);
}
