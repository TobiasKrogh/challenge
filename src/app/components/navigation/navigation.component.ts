import { ChangeDetectionStrategy, Component } from '@angular/core';
import { INavigationTarget } from '../../model/navigation-target.interface';

@Component({
  selector: 'challenge-navigation',
  templateUrl: './navigation.component.html',
  styleUrls: ['./navigation.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NavigationComponent {
  protected readonly navigationTargets: INavigationTarget[] = [
    { link: '/calculations', label: 'Calc', ref: 'calculations' },
    { link: '/apps', label: 'Apps', ref: 'apps' },
    { link: '/external', label: 'Ext', ref: 'external' },
    { link: '/reports', label: 'Reports', ref: 'reports' },
    { link: '/logout', label: 'Logout', ref: 'logout' },
  ];
}
