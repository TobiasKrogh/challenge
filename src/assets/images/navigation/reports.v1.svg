<svg xmlns="http://www.w3.org/2000/svg" width="25" height="25" fill="none"><path fill="#2DAEE5" d="M24.882 13.21h-12.11l-8.56 8.56a12.46 12.46 0 0 0 20.67-8.56ZM3.112 20.66A12.45 12.45 0 0 1 11.672 0v12.11l-8.56 8.56v-.01Z"/><path fill="#005B97" d="M13.232 0v11.65h11.65A12.44 12.44 0 0 0 13.232 0Z"/></svg>