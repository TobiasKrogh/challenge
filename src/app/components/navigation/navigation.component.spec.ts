import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { NavigationComponent } from './navigation.component';

describe('NavigationComponent', () => {
  let component: NavigationComponent;
  let fixture: ComponentFixture<NavigationComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [NavigationComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(NavigationComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  describe('When rendering', () => {
    it('Then it matches snapshot', () => {
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });
  });
});
