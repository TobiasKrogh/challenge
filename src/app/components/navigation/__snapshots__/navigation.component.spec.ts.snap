// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationComponent When rendering Then it matches snapshot 1`] = `
DebugElement {
  "nativeNode": <div
    id="root1"
  >
    <nav>
      <a
        class="navigation-link"
        href="/calculations"
      >
        <img
          alt="calculations"
          src="/assets/images/navigation/calculations.v1.svg"
        />
      </a>
      <a
        class="navigation-link"
        href="/apps"
      >
        <img
          alt="apps"
          src="/assets/images/navigation/apps.v1.svg"
        />
      </a>
      <a
        class="navigation-link"
        href="/external"
      >
        <img
          alt="external"
          src="/assets/images/navigation/external.v1.svg"
        />
      </a>
      <a
        class="navigation-link"
        href="/reports"
      >
        <img
          alt="reports"
          src="/assets/images/navigation/reports.v1.svg"
        />
      </a>
      <a
        class="navigation-link"
        href="/logout"
      >
        <img
          alt="logout"
          src="/assets/images/navigation/logout.v1.svg"
        />
      </a>
      
    </nav>
  </div>,
}
`;
